src/main/java/com/google/jenkins/plugins/computeengine/CleanLostNodesWork.java

```diff
@@ -86,27 +108,46 @@ private List<ComputeEngineCloud> getClouds() {
     }
 
     private Set<String> findLocalInstances(ComputeEngineCloud cloud) {
-        return Jenkins.get().getNodes().stream()
+        var localInstances = Jenkins.get().getNodes().stream()
                 .filter(node -> node instanceof ComputeEngineInstance)
                 .map(node -> (ComputeEngineInstance) node)
                 .filter(node -> node.getCloud().equals(cloud))
                 .map(Slave::getNodeName)
                 .collect(Collectors.toSet());
+        logger.log(Level.FINEST, "Found " + localInstances.size() + " local instances");
+        return localInstances;
     }
 
-    private List<Instance> findRemoteInstances(ComputeEngineCloud cloud) {
-        Map<String, String> filterLabel = ImmutableMap.of(CLOUD_ID_LABEL_KEY, cloud.getInstanceId());
+    private List<Instance> findRunningRemoteInstances(ComputeClientV2 clientV2) {
         try {
-            return cloud.getClient().listInstancesWithLabel(cloud.getProjectId(), filterLabel).stream()
-                    .filter(instance -> shouldTerminateStatus(instance.getStatus()))
-                    .collect(Collectors.toList());
+            var remoteInstances = clientV2.retrieveInstanceByLabelKeyAndStatus(NODE_IN_USE_LABEL_KEY, "RUNNING");
+            logger.log(Level.FINEST, "Found " + remoteInstances.size() + " running remote instances");
+            return remoteInstances;
         } catch (IOException ex) {
             logger.log(Level.WARNING, "Error finding remote instances", ex);
             return emptyList();
         }
     }
 
-    private boolean shouldTerminateStatus(String status) {
-        return !status.equals("STOPPING");
+    /**
+     * Updates the label of the local instances to indicate they are still in use. The method makes N network calls
+     * for N local instances, couldn't find any bulk update apis.
+     */
+    private void updateLocalInstancesLabel(
+            ComputeClientV2 clientV2, Set<String> localInstances, List<Instance> remoteInstances) {
+        var remoteInstancesByName =
+                remoteInstances.stream().collect(Collectors.toMap(Instance::getName, instance -> instance));
+        var labelToUpdate = ImmutableMap.of(NODE_IN_USE_LABEL_KEY, String.valueOf(System.currentTimeMillis()));
+        for (String instanceName : localInstances) {
+            if (!remoteInstancesByName.containsKey(instanceName)) {
+                continue;
+            }
+            try {
+                clientV2.updateInstanceLabels(remoteInstancesByName.get(instanceName), labelToUpdate);
```

Comment:
or perhaps more legibly (SpotBugs will thank you too)
```suggestion
            var remoteInstance = remoteInstancesByName.get(instanceName);
            if (remoteInstance == null) {
                continue;
            }
            try {
                clientV2.updateInstanceLabels(remoteInstance, labelToUpdate);
```

Reply:
that makes more sense, ty